libgadget/blackhole.c

```diff
@@ -160,6 +160,7 @@ void blackhole(void)
     tw_accretion->UseNodeList = 1;
     tw_accretion->query_type_elsize = sizeof(TreeWalkQueryBHAccretion);
     tw_accretion->result_type_elsize = sizeof(TreeWalkResultBHAccretion);
+    tw_accretion->tree = tree;
```

Comment:
I suggest the TreeWalk module takes a tree argument rather than setting the tree member of the TreeWalk structure this way we "formally decouple" the walking policy from the tree.

Reply:
That's going to be kind of a pain because I would have to pass it down as an argument to every function in the TreeWalk module...